R/db-sql.R

```diff
@@ -112,6 +114,31 @@ sql_translate_env.DBIConnection <- function(con) {
 
 #' @export
 #' @rdname db-sql
+sql_returning_cols <- function(con, cols, table, ...) {
+  if (is_empty(cols)) {
+    return(NULL)
+  }
+
+  rlang::check_dots_empty()
+  UseMethod("sql_returning_cols")
+}
+
+#' @export
+sql_returning_cols.DBIConnection <- function(con, cols, table, ...) {
+  returning_cols <- sql_named_cols(con, cols, table = table)
+
+  sql_clause("RETURNING", returning_cols)
+}
+
+sql_named_cols <- function(con, cols, table = NULL) {
+  nms <- names2(cols)
+  nms[nms == cols] <- ""
+
+  cols <- sql_table_prefix(con, cols, table)
+  cols <- set_names(cols, nms)
+  escape(ident_q(cols), collapse = NULL, con = con)
+}
+
 sql_random <- function(con) {
```

Comment:
Looks like this was accidentally moved up to `sql_returning_cols()`:

```suggestion
#' @export
sql_random <- function(con) {
```